### ISSUE TITLE
terraform_workspace_remote false-positive

### ISSUE THREAD

**Issue by acdha** [terraform-linters/tflint#761]
tflint 0.16 reports an error for any reference to `terraform.workspace` if you use a remote backend of any sort.

```hcl
environment   = lower(terraform.workspace)
```

The reasoning at https://github.com/terraform-linters/tflint/blob/v0.16.0/docs/rules/terraform_workspace_remote.md is incomplete: it is possible to use a remote backend for state storage but not execution. Assuming it's not feasible to query the active workspace type, the docs should probably mention the possibility of ignoring it.

## Version

```
TFLint version 0.16.0
```

**Comment by wata727** [comment]
Than you for opening the issue, @acdha.

Unfortunately, I'm not familiar with use-cases for remote backend, but if there are valid cases for using `terraform.workspace`, we may need to consider a workaround.

@bendrucker What do you think?

**Comment by acdha** [comment]
Basically in our case we have not certified Terraform Cloud for secrets storage so we’re running it in the local execution mode. I don’t know how common this is.

**Comment by bendrucker** [comment]
Hmm makes sense. Probably not very common overall but certainly valid. I think we just need to disable the rule by default so that users can opt in if they primarily use remote execution.

**Comment by acdha** [comment]
I'm correct in assuming that tflint doesn't do any network I/O, so it can't check the workspace config? I wonder whether this might be as simple as having the docs steer you in the direction of adding the ignore step.

**Comment by bendrucker** [comment]
Correct. That's a good idea, if you assume a significant majority of users use remote execution. Disabling the rule for everyone definitely makes it less effective.

**Comment by bendrucker** [comment]
@acdha Please take a look and let me know if you think this covers it well enough: https://github.com/terraform-linters/tflint/pull/767

It's not a perfect solution, particularly if you have a mix of local and remote execution modules that otherwise share TFLint config. For that, it seems like we might pursue a git-style config loading pattern where TFLint traverses upwards from `.` to `/` and then merges all the configs it finds.

**Comment by acdha** [comment]
@bendrucker Sorry for not getting back to you earlier, but that looks good. Thanks!
